posydon/popsyn/transient_select_funcs.py

```diff
@@ -0,0 +1,247 @@
+    Additional properties from the history or oneline can be added to the transient population.
+    The output dataframe contains the following columns:
+    - time : the time of the event
+    - metallicity : the metallicity of the event
+    - S1_m_disk_radiated : the mass of the disk radiated by the first star
+    - S2_m_disk_radiated : the mass of the disk radiated by the second star
+    
+    Columns with pre and post SN data:
+    - S1_mass : the mass of the first star
+    - S2_mass : the mass of the second star
+    - S1_spin : the spin of the first star
+    - S2_spin : the spin of the second star
+    - orbital_period : the orbital period of the binary
+    - eccentricity : the eccentricity of the binary
+
+    Parameters
+    ----------
+    history_chunk : pd.DataFrame
+        The history chunk of the population.
+    oneline_chunk : pd.DataFrame
+        The oneline chunk of the population.
+    formation_channels_chunk : pd.DataFrame (can be made optional)
+        The formation pathway chunk of the population.
+    S1_S2 : str
+        The star that will be the progenitor of the GRB. Can be either 'S1' or 'S2'.
+    
+    Returns
+    -------
+    pd.DataFrame
+        A DataFrame containing the transient population of LGRBs with the columns mentioned above.
+    
+    Example
+    -------
+    
+    >>> GRB_selection_wrapper(pop.history[0], pop.oneline[0], pop.formation_channels.loc[0])
+    >>> pop.create_transient_population(GRB_selection_wrapper, name='GRB_S1')
+    """
+    columns_pre_post  = ['orbital_period', 'eccentricity', 'S1_spin', 'S2_spin']
+    columns = []
+    oneline_columns = ['metallicity', 'S1_m_disk_radiated', 'S2_m_disk_radiated']
+   
+    if S1_S2 == 'S1':
+        indices_selection = oneline_chunk.index[oneline_chunk['S1_m_disk_radiated'] > 0.0].to_numpy()
+        oneline_chunk = oneline_chunk.drop(columns=['S2_m_disk_radiated'])
+    elif S1_S2 == 'S2':
+        indices_selection = oneline_chunk.index[oneline_chunk['S2_m_disk_radiated'] > 0.0].to_numpy()
+        oneline_chunk = oneline_chunk.drop(columns=['S1_m_disk_radiated'])
+    else:
+        raise ValueError('S1_S2 must be either S1 or S2')
+    # no events in this chunk
+    if len(indices_selection) == 0:
+        return pd.DataFrame()
+    # filter out the events that are not relevant for the LGRB formation
+    selection = history_chunk.loc[indices_selection]    
+    if S1_S2 == 'S1':
+        S_mask = (selection['S1_state'] == 'BH') & (selection['S1_state'] != 'BH').shift(1) & (selection['step_names'] == 'step_SN')
+    elif S1_S2 == 'S2':
+        S_mask = (selection['S2_state'] == 'BH') & (selection['S2_state'] != 'BH').shift(1) & (selection['step_names'] == 'step_SN')
+    
+    GRB_df_synthetic = pd.DataFrame(index=indices_selection)
+    # Pre and post SN data
+    post_SN_hist = selection[S_mask]
+    pre_mask = S_mask.shift(-1)
+    pre_mask.iloc[-1] = False
```

Comment:
You can combine the two lines for shifting and replacing the new entry by a single command, which makes use of a parameter of the shift function:
`pre_mask = S_mask.shift(-1, fill_value=False)`
Beside making the code shorter, this should make the code more efficient as well, because you don't need to touch the new entry twice.